corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -258,6 +265,53 @@ hqDefine("data_dictionary/js/data_dictionary", [
             return [];
         };
 
+        self.isActiveCaseTypeDeprecated = function () {
+            const activeCaseType = self.getActiveCaseType();
+            if (activeCaseType) {
+                return activeCaseType.deprecated;
+            }
+            return false;
+        };
+
+        self.activeCaseTypeModuleCount = function () {
+            const activeCaseType = self.getActiveCaseType();
+            if (activeCaseType) {
+                return activeCaseType.appCount;
+            }
+            return 0;
+        };
+
+        self.deprecateCaseType = function () {
+            self.deprecateOrRestoreCaseType(true);
+        };
+
+        self.restoreCaseType = function () {
+            self.deprecateOrRestoreCaseType(false);
+        };
+
+        self.deprecateOrRestoreCaseType = function (shouldDeprecate) {
+            let activeCaseType = self.getActiveCaseType();
+            if (!activeCaseType) {
+                return;
+            }
+
+            activeCaseType.deprecated = shouldDeprecate;
+            $("#deprecate-case-type-error").addClass("hidden");
+            $.ajax({
+                url: initialPageData.reverse('deprecate_or_restore_case_type', activeCaseType.name),
+                method: 'POST',
+                data: {
+                    'is_deprecated': shouldDeprecate
+                },
+                success: function () {
+                    window.location.reload(true);
+                },
+                error: function (error) {
+                    $("#deprecate-case-type-error").removeClass("hidden");
```

Comment:
You can use jQuery's [hide/show](https://www.w3schools.com/jquery/jquery_hide_show.asp) methods for this purpose without having to deal with the HTML attributes directly.